### DISCUSSION TITLE
PWM %-Statusanzeige

### DISCUSSION THREAD

**Discussion by Prideland** [camueller/SmartApplianceEnabler#593] [Allgemein]
Hallo,
Ist es irgendwie möglich, anstelle oder zusätzlich zur Ampel, sich anzeigen zu lassen mit wie viel Prozent der PWM-Ausgang zurzeit angesteuert wird?
Anlog dazu welche Stufe bei einem Stufenschalter gerade aktive ist?

**Comment by camueller** [comment]
Stufen und auch der PWM-Wert sind ja eigentlich "Übersetzungen" der Leistung, die vom Home Manager vorgegeben wurde.
Beide Werte sehe ich als "interne" Werte des jeweilgen Schalters. Und die Leistung wird ja bereits neben der Ampel angezeigt.

**Reply by Prideland** [reply]
Ich hab mir alles nochmal angeguckt, bei mir steht neben der Ampel keine Leistung. Es wird nur die verbleibende min. und max. Energie angezeigt (PWM-Schalter).

**Comment by camueller** [comment]
Ich habe mir das im Code mal angesehen. Du hast Recht, aktuell wird die Leistung nicht angezeigt.

Da ich selbst den PWM-Schalter und Stufenschalter nicht verwende, habe ich diese Features nicht so im Blick, wie bei Dingen, die ich selbst nutze. Wenn Du mir dabei hilfst, diese Features zu verbessern, wäre das super. 

Ich habe den Code angepasst, sodass (bei meinen Tests) die Leistung angezeigt wird. Hier ist eine modifizierte Test-Version der aktuellen 2.3.1 für Dich:
https://drive.google.com/file/d/12Gs3bmKZSQbHS8kyCrySz07sIdpOxTRr/view?usp=sharing

Ich habe den Verdacht, dass einige Dinge angezeigt werden, die für PWM-Schalter und Stufenschalter nicht so sinnvoll sind.
Deshalb würde ich Dich bitten, hier jeweils einen Screenshot der Ampel bei eingeschaltetem und ausgeschaltetem Zustand zu posten. Ich kann dann Anpassungen machen und Dir eine weitere Testversion bereitstellen.

**Comment by Prideland** [comment]
Danke für die Anpassung!
Ich habe den SAE aktualisiert und bis jetzt sieht alles gut aus. Zurzeit habe ich jedoch nur ein PWM-Schalter definiert.


![PWM-TEST](https://github.com/camueller/SmartApplianceEnabler/assets/138702513/6085c0e0-71a4-4f2d-bfba-7d22e1b93e5d)

**Comment by camueller** [comment]
Das sieht gut aus. Ich denke, dass man das so lassen kann. Falls Dir noch was auffällt, kannst Du Dich ja melden.

**Comment by Clarion30** [comment]
Morgen, könntest du das nochmal hochladen würde das auch gerne bei mir einfügen !!
Brauche das für einen Stufenschalter.

**Comment by camueller** [comment]
https://drive.google.com/file/d/1NTPOWaoLXTx6791zPMfi_D638JVB87re/view?usp=sharing

**Comment by Clarion30** [comment]
Danke
